slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,370 @@
+:::
+
+## recipes package
+
+extensible framework for pipeable sequences of feature engineering steps provides preprocessing tools to be applied to data
+
+::: {.incremental .highlight-last}
+- Modular + extensible
+- pipeable
+- Deferred evaluation
+- Isolates test data from training data
+- Can do things formulas can't
+:::
+
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+[recipe(tip ~ ., data = taxi_train)]{style="color: #CA225E;"} %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+<br>
+
+Start by calling `recipe()` to denote the data source and variables used
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ [step_unknown]{style="color: #CA225E;"}(all_nominal_predictors()) %>%  
+\ \ [step_dummy]{style="color: #CA225E;"}(all_nominal_predictors()) %>%  
+\ \ [step_zv]{style="color: #CA225E;"}(all_predictors()) %>%  
+\ \ [step_log]{style="color: #CA225E;"}(distance) %>%  
+\ \ [step_normalize]{style="color: #CA225E;"}(all_numeric_predictors())
+:::
+
+<br>
+
+specifying what actions to take by adding `step_*()`s
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_dummy([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_zv([all_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_log([distance]{style="color: #CA225E;"}) %>% 
+\ \ step_normalize([all_numeric_predictors()]{style="color: #CA225E;"})
+:::
+<br>
+
+using {tidyselect} and recipes specific selectors to denote affected variables
+
+## Using a recipe
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+rec_spec <- recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>% 
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+<br>
+
+recipes are can be used with {workflows} to "combine" it with a model
+
+<br>
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+wf_spec <- workflow() %>%  
+\ \ [add_recipe(rec_spec)]{style="color: #CA225E;"} %>%  
+\ \ add_model(linear_reg())
+:::
```

Comment:
I would put this on its own slide, to give it more visibility, since this is the main way we think recipes should be used

Reply:
Done!